proto/data.go

```diff
@@ -330,19 +329,34 @@ func (v *Value) SetInteger(i int64) {
 	v.Bytes = encoding.EncodeUint64(nil, uint64(i))
 }
 
-// GetInteger decodes an int64 value from the bytes field of the receiver.
-func (v *Value) GetInteger() int64 {
-	if v == nil || v.Bytes == nil {
-		return 0
+// GetInteger decodes an int64 value from the bytes field of the receiver. If
+// the bytes field is not 0 or 8 bytes in length an error will be returned.
+func (v *Value) GetInteger() (int64, error) {
+	if v == nil || len(v.Bytes) == 0 {
+		return 0, nil
+	}
+	const uint64Size = int(unsafe.Sizeof(uint64(0)))
```

Comment:
Again, my intention was not that the constant be computed from the real size of uint64; that part I don't feel strongly about.

I was imagining that the constant would live next to the encoding function. The encoding scheme may change in the future - that we currently use exactly 8 bytes to encode uint64 is an implementation detail that might change.